examples/step-44/step-44.cc

```diff
@@ -2001,16 +1998,18 @@ namespace Step44
       {
         fe_values_ref.reinit(cell);
 
-        PointHistory<dim> *lqph =
-          reinterpret_cast<PointHistory<dim>*>(cell->user_pointer());
-
-        Assert(lqph >= &quadrature_point_history.front(), ExcInternalError());
-        Assert(lqph <= &quadrature_point_history.back(), ExcInternalError());
+        // In contrast to that which was previously called for,
+        // in this instance the quadrature point data is specifically
+        // non-modifiable since we will only be accessing data.
+        const std::vector<std::shared_ptr<const PointHistory<dim> > > lqph =
+            const_cast<const CellDataStorage<typename Triangulation<dim>::cell_iterator,
+                                             PointHistory<dim> >&>(quadrature_point_history).get_data(cell);
+        Assert(lqph.size() == n_q_points, ExcInternalError());
 
         for (unsigned int q_point = 0; q_point < n_q_points; ++q_point)
           {
-            const double det_F_qp = lqph[q_point].get_det_F();
-            const double J_tilde_qp = lqph[q_point].get_J_tilde();
+            const double det_F_qp = lqph[q_point]->get_det_F();
+            const double J_tilde_qp = lqph[q_point]->get_J_tilde();
             const double the_error_qp_squared = std::pow((det_F_qp - J_tilde_qp),
                                                          2);
             const double JxW = fe_values_ref.JxW(q_point);
```

Comment:
This is outside the current patch, so I can't comment on these lines. But the bottom of this function looks like this:
```
     std::pair<double, double> error_dil;
     error_dil.first = std::sqrt(dil_L2_error);
     error_dil.second = vol_current / vol_reference;
 
     return error_dil;
```
While you're already here, why not write this as
```
  return std::make_pair (..., ...);
```
and save 3 lines of code.